pymatgen/io/openmm/generators.py

```diff
@@ -0,0 +1,422 @@
+                same atom ordering.
+            topology_file: Location to save the Topology PDB.
+            system_file: Location to save the System xml.
+            integrator_file: Location to save the Integrator xml.
+            state_file: Location to save the State xml.
+        """
+        self.force_field = force_field
+        self.temperature = temperature
+        self.step_size = step_size
+        self.friction_coefficient = friction_coefficient
+        self.partial_charge_scaling = partial_charge_scaling if partial_charge_scaling else {}
+        self.partial_charges = partial_charges if partial_charges else []
+        self.packmol_random_seed = packmol_random_seed
+        self.topology_file = topology_file
+        self.system_file = system_file
+        self.integrator_file = integrator_file
+        self.state_file = state_file
+
+    def get_input_set(  # type: ignore
+        self,
+        smiles: Dict[str, int],
+        density: Optional[float] = None,
+        box: Optional[List[float]] = None,
+    ) -> InputSet:
+        """
+        This executes all of the logic to create the input set. It generates coordinates, instantiates
+        the OpenMM objects, serializes the OpenMM objects, and then returns an InputSet containing
+        all information needed to generate a simulaiton.
+
+        Please note that if the molecules are chiral, then the SMILEs must specify a
+        particular stereochemistry.
+
+        Args:
+            smiles: keys are smiles and values are number of that molecule to pack
+            density: the density of the system. density OR box must be given as an argument.
+            box: list of [xlo, ylo, zlo, xhi, yhi, zhi]. density OR box must be given as an argument.
+
+        Returns:
+            an OpenMM.InputSet
+        """
+        assert (density is None) ^ (box is None), "Density OR box must be included, but not both."
+        smiles = {smile: count for smile, count in smiles.items() if count > 0}
+        # create dynamic openmm objects with internal methods
+        topology = self._get_openmm_topology(smiles)
+        if box is None:
+            box = self.get_box(smiles, density)  # type: ignore
+        coordinates = self._get_coordinates(smiles, box, self.packmol_random_seed)
+        smile_strings = list(smiles.keys())
+        system = self._parameterize_system(
+            topology,
+            smile_strings,
+            box,
+            self.force_field,
+            self.partial_charge_scaling,
+            self.partial_charges,
+        )
+        integrator = LangevinMiddleIntegrator(
+            self.temperature * kelvin,
+            self.friction_coefficient / picoseconds,
+            self.step_size * picoseconds,
+        )
+        context = Context(system, integrator)
+        context.setPositions(coordinates)
+        state = context.getState(getPositions=True)
```

Comment:
Creating this `Context` caused me a truly nightmarish bug when trying to run on the Perlmutter GPUs. Somehow, instantiating a `Context` creates some state in the program that persists even after `get_input_set` has run. Later on, when I call `input_set.get_simulation`, the `Simulation` internally try's to create a `Context` that conflicts with the previous `Context`, causing an error and crashing the program. This does not occur locally when I use `OpenCL` or `CPU` as the platform, and it does not occur when I run remotely using `CPU`. Absolute nightmare.

Unfortunately, we can't instantiate a `State` without creating a `Context`, so these lines are integral to the current IO implementation.

There is no issue if we create the input files in one python file and then load them in another, it is only an issue if we create and load them in the same file. So thankfully, this shouldn't be too much of an issue for HTMD. 

I'm tempted to just ignore this issue and make sure the error is well documented. It's enough of an edge case that I don't think it warrants restructuring the IO. Curious to hear what you think though.

This was awful to debug, so I'm thankful to have gotten to the bottom of this error.